### ISSUE TITLE
Troube interfacing with custom pass script

### ISSUE THREAD

**Issue by TheCraiggers** [nwallace/passff#156]
This is a bit of a weird request, I apologize if this is a waste of your time. I was dissatisfied with the options for using pass on Windows machines. The only two I could find were GUI options. I wanted something on the command line so it was more compatible with tools like PassFF. I also wanted the least amount of dependencies. So I'm making something small to do the trick.

I have a working set of scripts [here](https://github.com/TheCraiggers/winpass) if you're curious. It's rough, but I wanted to start small and work from there. Of course, I've already hit a snag.

PassFF is able to get the password list just fine. However, when it comes to decrypting a password, nothing seems to happen. In the browser console, I see the following:

```
[PassFF] Execute pass Object { command: "d:\passwds\winpass\pass.bat", arguments: Array[1], environment: Array[8], charset: "UTF-8", mergeStderr: false, done: PassFF.Pass.executePass/params.done() }
[PassFF] [subprocess] trying to close stdin
[PassFF] [subprocess] Closing Stdin
[PassFF] [subprocess] got 53 bytes from stderr
[PassFF] [subprocess] got 86 bytes from stderr
[PassFF] [subprocess] Got msg from stdout: EOF reached
[PassFF] [subprocess] Got msg from stderr: EOF reached
[PassFF] [subprocess] Pipe stderr closed
[PassFF] [subprocess] Pipe stdout closed
[PassFF] [subprocess] Cleanup called
[PassFF] [subprocess] trying to close stdin
[PassFF] pass script execution ok
```

Obviously, the two lines from stderr doesn't look good, and is probably causing the issue. However, if I run the pass command manually from the command line, everything works like a charm. PassFF does seem to be calling the script correctly, and passing the needed parameter. So I'm at a loss for what the issue could be, unless PassFF just totally expects a *NIX type environment and doesn't know exactly how to handle Windows in this case. I'm hoping somebody here can help point me in a direction. Here is an example from the command line in case something seems off about it that I haven't seen:

```
d:\Passwds\winpass>pass amazon
afjkah74hlashfaksjfh74alifsudfkajshfkajsafhg74alsuhfjkhf

URL: www.amazon.com
Login: loginname
```

**Comment by TheCraiggers** [comment]
I've done some additional debugging, and I have verified at PassFF is calling the script and passing the name of the password file correctly. But it really hates the output. Is there an easy way of seeing what is being captured? I've not found a way to do it in the console.

**Comment by phryneas** [comment]
Not unless you enable it in the code and make the plugin for yourself - this is for your own protection.

But just as a question: why don't you just run the original pass with cygwin or msys2 (I recommend the latter)? It works like a charm.

Fellow windows user here ;)

**Comment by TheCraiggers** [comment]
Well, originally I didn't want to use them because I felt I could easily make something to emulate pass. I mean, it's just a file list and a call to gpg, basically. I figured I could get the important parts working with only a few lines in a batch file, with no extra dependencies. And I was right, it works fine. It just doesn't work with PassFF.

Oh well, I'm out of time for this pet project, so I figure I'll probably take your advice and try out msys2. I've used cygwin a few times, but never msys2. Were there any pitfalls you had getting it set up with pass?
